src/main/java/org/opentripplanner/routing/algorithm/RoutingWorker.java

```diff
@@ -63,29 +65,53 @@ public RoutingWorker(RaptorConfig<TripSchedule> config, RoutingRequest request)
     }
 
     public RoutingResponse route(Router router) {
-        try {
-            List<Itinerary> itineraries;
+        List<Itinerary> itineraries = new ArrayList<>();
+        List<RoutingError> routingErrors = new ArrayList<>();
```

Comment:
I would make the routingErrors a member of the request scoped Worker class - then each of the two try/catch blocks can be moved into their own methods and don´t interrupt the reading of the main flow. This have the side-effect of changing the state of the worker, so that should be documented for each of the methods.

Reply:
As you suggested in another comment, I've instead opted to use a general RoutingValidationException that includes a RoutingError.